### PR TITLE
docs: Update gsoc timeline

### PR DESCRIPTION
**Pull request by Xuanwo**
I'm not sure about this change. Are we stick to this timeline? Or we just remove them entirely?

Current timeline is a bit confusing.

### CODE REVIEW

**Comment by Xuanwo** [comment]
cc priyasharma1, would you like to take a review on this? Thanks!

File: source/gsoc/_index.md [hunk 1]
```diff
@@ -136,35 +136,35 @@ date indicated.
 <a name="GSoC-StudentApplicationandSelection"></a>
 ## Student Application and Selection
 
-  - 2023-02-22: Potential GSoC contributors discuss application ideas with mentoring organizations
```
Comment (solomax) [inline source/gsoc/_index.md:139]: the dates are different for each year

plus we are shifting original GSoC dates (making them 1-2 days earlier) to have enough time to ping mentors ....

Comment (Xuanwo) [reply source/gsoc/_index.md:139]: How about remove the old dates and leave a TBD here?

**Review by solomax — CHANGES_REQUESTED** [review]
I would have hardcoded year and dates from https://developers.google.com/open-source/gsoc/timeline shifter by two days for `evaluation deadlines`

File: source/gsoc/_index.md (hunk 1 above)
Comment (solomax) [reply source/gsoc/_index.md:139]: well, I would VOTE for having up-to-date dates (to have something for the reference)

**Comment by Xuanwo** [comment]
> I would have hardcoded year and dates from [developers.google.com/open-source/gsoc/timeline](https://developers.google.com/open-source/gsoc/timeline) shifter by two days for `evaluation deadlines`

Updated, PTAL.

**Review by solomax — APPROVED** [review]

**Comment by solomax** [comment]
Thanks a million @Xuanwo !

**Pull request merged**